zio/anyio/mime.go

```diff
@@ -0,0 +1,74 @@
+package anyio
+
+import (
+	"fmt"
+	"io"
+	"mime"
+
+	"github.com/brimdata/zed/zio"
+	"github.com/brimdata/zed/zio/jsonio"
+	"github.com/brimdata/zed/zio/ndjsonio"
+	"github.com/brimdata/zed/zio/zjsonio"
+	"github.com/brimdata/zed/zio/zngio"
+	"github.com/brimdata/zed/zio/zsonio"
+	"github.com/brimdata/zed/zson"
+)
+
+const (
+	MimeTypeCSV    = "application/csv"
+	MimeTypeJSON   = "application/json"
+	MimeTypeNDJSON = "application/ndjson"
+	MimeTypeZJSON  = "application/x-zjson"
+	MimeTypeZSON   = "application/x-zson"
+	MimeTypeZNG    = "application/x-zng"
+)
+
+func WriterByMimeType(w io.WriteCloser, v string) (zio.WriteCloser, string, error) {
+	if v == "" {
+		v = MimeTypeJSON
+	}
+	typ, _, err := mime.ParseMediaType(v)
+	if err != nil {
+		return nil, "", err
+	}
+	switch typ {
+	case MimeTypeJSON:
+		return jsonio.NewWriter(w), MimeTypeJSON, nil
+	case MimeTypeNDJSON:
+		return ndjsonio.NewWriter(w), MimeTypeNDJSON, nil
+	case MimeTypeZJSON:
+		return zjsonio.NewWriter(w), MimeTypeZJSON, nil
+	case MimeTypeZSON:
+		return zsonio.NewWriter(w, zsonio.WriterOpts{}), MimeTypeZSON, nil
+	case MimeTypeZNG:
+		return zngio.NewWriter(w, zngio.WriterOpts{}), MimeTypeZNG, nil
+	default:
+		return nil, "", fmt.Errorf("unknown mime type: %s", typ)
+	}
```

Comment:
Maybe extract this stuff? And ignore `mime.ErrInvalidMediaParameter`?
Then you can just call `anyio.NewReaderWithOpts` or `anyio.NewWriter`, setting `Format` in the options parameter.
```go
function mediaTypeToFormat(s string) (string, error) {
	typ, _, err := mime.ParseMediaType(s)
	if err != nil && err != mime.ErrInvalidMediaParameter {
		return nil, err
	}
	switch typ {
	case MimeTypeJSON:
		return "json", nil
	case MimeTypeNDJSON:
		return "ndjson", nil
	case MimeTypeZJSON:
		return "ndjson", nil
	case MimeTypeZNG:
		return "zng", nil
	case MimeTypeZSON:
		return "zson", nil
	}
	return "", fmt.Errorf("unknown MIME type: %s", typ)
}
```

Reply:
Specifically,
```go
	format, err := mediaTypeToFormat(resp.Header().Get("Content-Type"))
	if err != nil {
		return nil, err
	}
	return anyio.NewReaderWithOpts(r, zctx, anyio.ReaderOpts{
		Format: format,
	})
```